File: Shortest_Path_Algo/2021-remove-all-occurrences-of-a-substring/remove-all-occurrences-of-a-substring.cpp
class Solution {
public:
    string removeOccurrences(string s, string part) {
        string st = "";
        int partSize = part.size();

        for(char ch:s){
            st += ch;

            if(st.size() >= partSize && st.substr(st.size() - partSize) == part)
                st.erase(st.size() - partSize);
        }

        return st;
    }
};

File: Shortest_Path_Algo/2021-remove-all-occurrences-of-a-substring/remove-all-occurrences-of-a-substring.py
class Solution:
    def removeOccurrences(self, s: str, part: str) -> str:
        st = ""
        part_size = len(part)

        for ch in s:
            st += ch
            if len(st) >= part_size and st[-part_size:] == part:
                st = st[:-part_size]

        return st
